### PR TITLE
171784032 Signup to Barefoot Nomad UI

### PR DESCRIPTION
**Pull request by Francois-MUGOROZI**
### What the PR does?
User get registration form to signup 
### Description of tasks to be completed
 - add signup form
 - connect with API service to signup new user
### How this should be tested manually?
having node/npm, clone the repo branch, cd into and run "npm install".
Then run "start:dev", the browser will open and navigate to signup

### Relevent Pivotal tracker story
[171784032](https://www.pivotaltracker.com/n/projects/2438874/stories/171784032)

### Screenshoots
**Desktop**
Screenshot (61)

**Mobile**
Screenshot (62)

### CODE REVIEW

File: src/actions/signupAction.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import axios from 'axios';
+
+const signupHandle = async (signupData) => {
+  try {
+    const res = await axios.post(`http://localhost:5000/api/v1/users/signup`, {
```
Comment (benshidanny11) [inline src/actions/signupAction.js]: Great job so far, but however I think you should try to use Heroku URL instead of the Local one.

**Review by benshidanny11 — CHANGES_REQUESTED** [review]

File: src/components/auth/Signup.js [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    const signupData = {
+      firstname: this.state.firstname,
+      lastname: this.state.lastname,
+      email: this.state.email,
+      password: this.state.password,
+      confirmpassword: this.state.confirmpassword,
+    };
+    if (this.state.error) {
+    } else {
+      signupData.confirmpassword = undefined;
```
Comment (bdushimi) [inline src/components/auth/Signup.js]: @Francois-MUGOROZI any reason why you're making this property `confirmpassword` undefined?

File: src/constants/urls.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default {
+  API_BASE_URL: 'http://localhost:5000/api/v1',
```
Comment (bdushimi) [inline src/constants/urls.js]: @Francois-MUGOROZI shouldn't this point to heroku backend APIs?

File: src/tests/redux/actions/signup.test.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    email: 'username@gmail.com',
+    password: 'password123',
+  };
+  afterEach(() => {
+    fetchMock.restore();
+  });
+  it('SHould create Signup action', () => {
+    fetchMock.postOnce('/signup', data);
+    const store = mockStore({ signup: {} });
+    const expectedActions = [{ type: SIGNUP_STARTED }];
```
Comment (bdushimi) [inline src/tests/redux/actions/signup.test.js:50]: @Francois-MUGOROZI The signup function/thunk fires about more that one action, you have also to assertain signupSucceded and signupFailed actions too as a result, all lines shall be covered.

**Review by bdushimi — CHANGES_REQUESTED** [review]
1. Once a user visits the signup page, there's no need of keeping/including the signup button on the signup page. I suggest you remove the signup button
2. There's a typo right here `By signimg up`. Kindly fix that.
3. I would be better to have the title above the signup page, taking the remaining width. Instead of it being like this :
`Signup To
  Barefoot Nomad`
Make it look like this Signup To Barefoot Nomad
4. When a user submits the data, can you inactivate the form while waiting for the API response?
5. If an API request fails, the form/page does not show any type of error message of what happened/went wrong. (Note that the error message should be user friendly).
6.Kindly ensure that form validations are handled correctly (incorrect emails, not matching passwords)
7. Regarding the tests, some lines in the signup component are not covered because you did not initialize the state so that all properties have defined values.
8. 

Good job on responsiveness and overall implementation.

File: src/assets/styles/auth/common.scss [hunk 5]
```diff
@@ -0,0 +1,38 @@
+.app {
+  display: flex;
+  min-height: 100vh;
+  flex-direction: column;
+}
+main {
+  flex: 1 0 auto;
+}
+
+.auth-btn {
```
Comment (irfiacre) [inline src/assets/styles/auth/common.scss]: I think it would better to separate the `styles`  and the `assets`.

File: src/index.js [hunk 6]
```diff
@@ -1,5 +1,13 @@
...
+// Materialize CSS setup
+import 'materialize-css/dist/css/materialize.min.css';
+import 'font-awesome/css/font-awesome.min.css';
+
+// custom styles
+import './assets/styles/index.scss';
 import App from './App'; // Application setup
 
-ReactDOM.render(App, window.document.getElementById('root'));
+reactDOM.render(<App/>, window.document.getElementById('root'));
```
Comment (irfiacre) [inline src/index.js]: Why did you change this?

**Review by irfiacre — CHANGES_REQUESTED** [review]
Good job @Francois-MUGOROZI this is good.
But from what I have seen in your implementation of the signup UI there are some things that do not resemble the initial design:
image

From your UI:

image
+(inputs)
Why?

File: src/index.js (hunk 6 above)
Comment (Francois-MUGOROZI) [reply src/index.js]: In my end it was not working that way

**Comment by bdushimi** [comment]
@Francois-MUGOROZI Kindly reduce the height of both the picture & footer in order to avoid scrolling on the signup page. No need to scroll on the signup page.

**Comment by bdushimi** [comment]
@Francois-MUGOROZI The input text errors should appear once a user has finished typing in there and the input text is not onfocus.

**Comment by bdushimi** [comment]
@Francois-MUGOROZI Regarding password & confirm password input texts, let them have different validations, also change this error message 'Password confirm must match password', it's kind confusing, I suggest you go with "Passwords must match".

**Comment by bdushimi** [comment]
@Francois-MUGOROZI I think it would look better if you reduce the width of the button "Create Free Account"

**Comment by bdushimi** [comment]
Screen Shot 2020-05-29 at 11 54 37

There's a CORS error, I am getting.

**Comment by bdushimi** [comment]
@Francois-MUGOROZI let this title "Signup To Barefoot Nomad" be centered as well.

**Review by irfiacre — APPROVED** [review]

File: src/services/example.service.js [hunk 7]
```diff
@@ -0,0 +1 @@
+// carryout the first action
```
Comment (irfiacre) [inline src/services/example.service.js:1]: May be you could delete this file

**Review by irfiacre — APPROVED** [review]

**Review by bdushimi — APPROVED** [review]

**Pull request merged**